pennylane/labs/trotter_error/product_formulas/error.py

```diff
@@ -302,13 +299,13 @@ def _apply_commutator(
 
         new_state += coeff * tmp_state
 
-    return new_state
+    return state.dot(new_state)
 
 
-def _apply_commutator_track_order(
+def _compute_expectation_track_order(
     commutator: tuple[Hashable], fragments: dict[Hashable, Fragment], state: AbstractState
-) -> tuple[AbstractState, int]:
-    """Returns the state obtained from applying ``commutator`` to ``state``."""
+) -> tuple[complex, int]:
+    """Returns the expectation value obtained from applying ``commutator`` to ``state``."""
```

Comment:
Was it just a typo or the logic of the function is changed? Also, there was no test to catch this?